HW2/P3/mandelbrot.pyx

```diff
@@ -0,0 +1,148 @@
+import numpy as np
+cimport numpy as np
+cimport cython
+import numpy
+cimport AVX
+from cython.parallel import prange
+
+#We adapt the magnitude_squared function for an AVX.float8 element (we follow magnitude_squared = z_real^2 + z_imag^2)
+cdef AVX.float8 magnitude_squared(AVX.float8 z_real, AVX.float8 z_imag) nogil:
+    return AVX.fmadd(z_real, z_real, AVX.mul(z_imag, z_imag))
+
+
+#We create a function to copy the temporary tmp_counts buffer to the out_counts array. 
+cdef void counts_to_output(AVX.float8 counts,
+                      np.uint32_t [:, :] out_counts,
+                      int i, int j) nogil:
+    
+    cdef:
+        float tmp_counts[8]
+        int jj
+        
+    AVX.to_mem(counts, &(tmp_counts[0])) 
+    
+    for jj in range(8):
+        out_counts[i, 8*j+jj] =  <int>tmp_counts[jj]
```

Comment:
This is a good way of adding the counts to the output, I did something slightly different that was much more confusing. This is a cleaner, easier to follow method. Well done!